site-manage/site-manage-impl/impl/src/java/org/sakaiproject/sitemanage/impl/SiteManageServiceImpl.java

```diff
@@ -794,4 +809,62 @@ private void updateSiteInfoToolEntityReferences(Map<String, String> transversalM
             }
         }
     }
+
+    /**
+     * Copy tool permissions from source site to destination site
+     * @param fromSiteId The source site ID
+     * @param toSiteId The destination site ID
+     */
+    private void copyToolPermissions(String fromSiteId, String toSiteId) {
+        try {
+            // Get the source site
+            log.info("Starting tool permissions copy from site {} to site {}", fromSiteId, toSiteId);
+            Site fromSite = siteService.getSite(fromSiteId);
+            // Get the destination site
+            Site toSite = siteService.getSite(toSiteId);
+            
+            // Copy all role permissions from source site to destination site
+            Set<Role> fromRoles = fromSite.getRoles();
+            log.debug("Found {} roles in source site", fromRoles.size());
+            
+            for (Role fromRole : fromRoles) {
+                String roleName = fromRole.getId();
+                try {
+                    // Get the corresponding role in the destination site
+                    Role toRole = toSite.getRole(roleName);
+                    if (toRole != null) {
+                        // Get all permissions from the source role
+                        Set<String> allPermissions = fromRole.getAllowedFunctions();
+                        log.debug("Copying {} permissions for role {}", allPermissions.size(), roleName);
+                        
+                        // Copy all permissions (don't filter by tool)
+                        for (String permission : allPermissions) {
+                            // If the source role has this permission, add it to the destination role
+                            if (fromRole.isAllowed(permission)) {
+                                toRole.allowFunction(permission);
+                                log.debug("Allowing permission {} for role {}", permission, roleName);
+                            } else {
+                                // If the permission was explicitly disallowed, disallow it in the destination role
+                                toRole.disallowFunction(permission);
+                                log.debug("Disallowing permission {} for role {}", permission, roleName);
+                            }
+                        }
+                    } else {
+                        log.warn("Role {} not found in destination site {}", roleName, toSiteId);
+                    }
+                } catch (Exception e) {
+                    log.warn("Could not find role {} in destination site {}", roleName, toSiteId);
+                }
+            }
+            
+            // Save the destination site with the updated permissions
+            siteService.save(toSite);
```

Comment:
does this actually save the new permissions? I thought siteService only saved the site structure and not the permissions (authzService)

Reply:
yes it does save it; https://github.com/sakaiproject/sakai/blob/8eb5b80e86cc5cda34009765563095584a585926/kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseSiteService.java#L1030